manifests/0000_90_etcd-operator_03_prometheusrule.yaml

```diff
@@ -150,43 +138,28 @@ spec:
         runbook_url: https://github.com/openshift/runbooks/blob/master/alerts/cluster-etcd-operator/etcdHighNumberOfFailedGRPCRequests.md
         summary: etcd cluster has high number of failed grpc requests.
       expr: |
-        100 * sum(rate(grpc_server_handled_total{job="etcd", grpc_code=~"Unknown|FailedPrecondition|ResourceExhausted|Internal|Unavailable|DataLoss|DeadlineExceeded"}[5m])) without (grpc_type, grpc_code)
+        (sum(rate(grpc_server_handled_total{job="etcd", grpc_code=~"Unknown|FailedPrecondition|ResourceExhausted|Internal|Unavailable|DataLoss|DeadlineExceeded"}[5m])) without (grpc_type, grpc_code)
           /
-        sum(rate(grpc_server_handled_total{job="etcd"}[5m])) without (grpc_type, grpc_code)
-          > 5 and on ()(sum(cluster_infrastructure_provider{type!~"ipi|BareMetal"} == bool 1))
+        (sum(rate(grpc_server_handled_total{job="etcd"}[5m])) without (grpc_type, grpc_code)
+          > 2 and on ()(sum(cluster_infrastructure_provider{type!~"ipi|BareMetal"} == bool 1)))) * 100 > 50
```

Comment:
Changing from `5` to `2` and adding `> 50` both seem like semantic changes, but there is no `description` update.  It also seems a fair bit different than [the upstream `expr`][1].  I haven't rooted around in history to figure out what's going on here, but I'm curious ;).

[1]: https://github.com/etcd-io/etcd/blame/08f4c340eb92338d0f266795222d9ea04cb69056/contrib/mixin/mixin.libsonnet#L90-L123

Reply:
I have to be honest, this alert could've never worked in Prometheus. So I think we're going to contribute this back upstream soon after we unblocked the SDN folks with this. 

> Changing from 5 to 2 and adding > 50 both seem like semantic changes, but there is no description update.

5 and 2 are supposed to be a low-pass filter, filtering out the low-QPS endpoints that can have high failure rates without impacting the stability. As to where those numbers come from? No idea, but it seems two QPS is good enough to filter the least called ones out.

The description as `etcd cluster "{{ $labels.job }}": {{ $value }}% of requests for {{ $labels.grpc_method }} failed on etcd instance {{ $labels.instance }}.` is good enough IMHO, any changes you would like to see here?